### DISCUSSION TITLE
How to ensure a Remark syntax extension is compatible with remark-cli?

### DISCUSSION THREAD

**Discussion by benrbray** [remarkjs/remark#980] [Q&A] [answered]
I'm the author of [`@benrbray/remark-cite`](https://github.com/benrbray/remark-cite) and I recently received [an issue](https://github.com/benrbray/remark-cite/issues/1) from someone hoping to use my syntax extension with `remark-cli`.  Unfortunately I was unable to track down the source of the problem, mostly because I had trouble figuring out how a plugin declared in the config file is actually loaded during execution of `remark-cli`.

Any advice for debugging the issue?  The extension works fine when passed directly to `remark` as a plugin.

**Comment by wooorm** [comment, marked as answer]
It should just work with everything. The CLI passes some extra information that plugins could use, which you found, but that's not a problem.
- you can make sure the versions match (remark --version)
- your plugin is very similar  to remark-frontmatter, remark-gfm, etc, so I don't see why it wouldn't work
- it seems weird that this only fails on the CLI, I don't see a reason for that to be the case 🤷‍♂️
- maybe running with debug (DEBUG='*' remark...) gives some more insight?
